subprojects/core-api/src/main/java/org/gradle/api/artifacts/Configuration.java

```diff
@@ -543,6 +543,24 @@ public String determineName(Configuration c) {
      */
     boolean isCanBeResolved();
 
+    /**
+     * Configures if a configuration can have dependencies declared upon it.
+     *
+     * @since 8.1
+     */
+    @Incubating
+    void setCanBeDeclaredAgainst(boolean allowed);
```

Comment:
I've put some prior thought into these names. I think that `is/setCanBeDeclaredUpon` would be more fitting, since the phrase "declare a dependency on <configuration name>" is the preferred terminology as opposed to "declare a dependency against <configuration name>"

Thoughts here? Its not a big deal and I don't think we need to deal with this here (since it would imply changes to tests and other existing first-party plugin implementations). But I thought I'd bring this up since we're now exposing this as part of a public API.

Reply:
I talked with Sterling about this, and I'm going to try a couple of things to avoid adding to the public Configuration API.  I think the name will matter less then.

But I have no strong feelings about the name.  If renamin isn't a pain, I'll include it in this change.  If it is, I'll do it as part of the series of changes after this.